lib/cluster/strategy/postgres.ex

```diff
@@ -52,24 +52,19 @@ defmodule Cluster.Strategy.Postgres do
   end
 
   def handle_continue(:connect, state) do
-    with {:ok, conn} <- Postgrex.start_link(state.meta.opts.()),
-         {:ok, conn_notif} <- Postgrex.Notifications.start_link(state.meta.opts.()),
-         {_, _} <- Postgrex.Notifications.listen(conn_notif, state.config[:channel_name]) do
-      Logger.info(state.topology, "Connected to Postgres database")
-
-      meta = %{
-        state.meta
-        | conn: conn,
-          conn_notif: conn_notif,
-          heartbeat_ref: heartbeat(0)
-      }
-
-      {:noreply, put_in(state.meta, meta)}
-    else
-      reason ->
-        Logger.error(state.topology, "Failed to connect to Postgres: #{inspect(reason)}")
```

Comment:
What happens when this fails loudly does it take down the whole app with it?

Reply:
I would need to check exactly, but in theory, if the failure happen repeatedly, then it can cause whole application to go down.

Reply:
> does it take down the whole app with it?


No, Supervisor will try to restart the Postgres strategy process, and after exhausting the retries, it will just leave it unstarted. But that change doesn’t make much sense anyway, because the node will be out of the cluster in case of a Postgres failure. You probably want to forward already assigned values, send a connect message after a timeout, and handle them more gracefully